### DISCUSSION TITLE
Sanitizer to allow base64 images?

### DISCUSSION THREAD

**Discussion by andreisctendava** [mganss/HtmlSanitizer#280] [Q&A]
Hi, 

I have a question.

How I can configure the Sanitizer to allow base64 images 
`<img src=\"data:image/png;base64, iVBORw0KGgo...">` 
and  disallow images like this 
`<img src="url_to_image">`?

Thanks.

**Comment by tiesont** [comment]
See https://github.com/mganss/HtmlSanitizer/wiki/Examples#ex2-allowing-data-uris for an example of how to allow data URIs. I'd assume you could extend than relatively easily.

There's also the [FilterUrl](https://github.com/mganss/HtmlSanitizer/wiki/Hooks) event. You'd likely need to implement a custom event handler for that, assuming your content can have HTTP/HTTPS URLs for other elements, like anchor tags. If not, you could clear the AllowedSchemes collection and then add "data".
